### PR TITLE
Add `UrlConstraints.__hash__`

### PR DESCRIPTION
**Pull request by madkinsz**
## Change Summary

Adds `__hash__` implementation to `UrlConstraints`.

## Related issue number

Closes https://github.com/pydantic/pydantic/issues/5693

## Checklist

* [x] Unit tests for the changes exist
* [x] Tests pass on CI and coverage remains at 100%
* [x] Documentation reflects the changes where applicable
* [x] `changes/<pull request or issue id>-<github username>.md` file added describing change
  (see [changes/README.md](https://github.com/samuelcolvin/pydantic/blob/master/changes/README.md) for details)
* [x] My PR is ready to review, **please add a comment including the phrase "please review" to assign reviewers**

Selected Reviewer: @hramezani

### CODE REVIEW

**Comment by madkinsz** [comment]
please review

File: changes/5693-madkinsz.md [hunk 1]
```diff
@@ -0,0 +1 @@
+Adds `__hash__` to `UrlConstraints` allowing it to be used as a field annotation with `Union`.
```
Comment (hramezani) [inline changes/5693-madkinsz.md:1]: As we are preparing pydantic for `V2` and there are a lot of changes, I think we don't need this file

**Comment by hramezani** [comment]
please add a test that shows using `HttpUrl` in model won't raise the error in the issue

**Comment by hramezani** [comment]
please update

File: changes/5693-madkinsz.md (hunk 1 above)
Comment (madkinsz) [reply changes/5693-madkinsz.md]: Removed

**Comment by madkinsz** [comment]
> please add a test that shows using `HttpUrl` in model won't raise the error in the issue

This is already done; see https://github.com/pydantic/pydantic/pull/5697/files#diff-f953eed7c7855cc439287af4864f147a1ed9c36995d18f335b57b96e3d7653b3R278-R283 / https://github.com/pydantic/pydantic/pull/5697/commits/50fbea8e96977280b26bfa466c6744ad28671869

File: tests/test_networks.py [hunk 2]
```diff
@@ -647,6 +656,60 @@ class Model2(BaseModel):
...
+        {'default_path': 'new-path'},
+        {'max_length': None},
+        {'allowed_schemes': None},
+        {'host_required': None},
+        {'default_host': None},
+        {'default_port': None},
+        {'default_path': None},
+    ],
+)
+def test_url_constraints_hash_inequal(changes: dict):
```
Comment (hramezani) [inline tests/test_networks.py:701]:
```suggestion
def test_url_constraints_hash_inequal(changes: Dict[str, Any]):
```

**Review by samuelcolvin — APPROVED** [review]
I've made some tweaks, now good.

**Comment by samuelcolvin** [comment]
This is great, thanks so much @madkinsz.

**Pull request merged**